test/integration/update_latest_snapshot_test.go

```diff
@@ -127,6 +131,75 @@ func (s *UpdateLatestSnapshotSuite) TestUpdateLatestSnapshot() {
 	assert.NoError(s.T(), err)
 }
 
+func (s *UpdateLatestSnapshotSuite) TestUpdateLatestSnapshotForRedHatRepo() {
+	config.Get().Features.Snapshots.Enabled = true
+	err := config.ConfigureTang()
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), config.Tang)
+
+	s.dao = dao.GetDaoRegistry(db.DB)
+	ctx := context.Background()
+	orgID := uuid2.NewString()
+
+	// Create a "red hat" repository and add it to a template
+	url, cancelFunc, err := ServeRandomYumRepo(nil)
+	require.NoError(s.T(), err)
+	defer cancelFunc()
+
+	repoResp, err := s.rhelRepo(url, config.SubscriptionFeaturesIgnored[0])
+	require.NoError(s.T(), err)
+
+	// Start the task
+	taskClient := client.NewTaskClient(&s.queue)
+	uuidStr, err := uuid2.Parse(repoResp.RepositoryUUID)
+	require.NoError(s.T(), err)
+
+	s.snapshotAndWait(taskClient, *repoResp, uuidStr, true)
+
+	// Create template with use latest
+	reqTemplate := api.TemplateRequest{
+		Name:            utils.Ptr(fmt.Sprintf("test template %v", rand.Int())),
+		Description:     utils.Ptr("includes rpm unsigned"),
+		RepositoryUUIDS: []string{repoResp.UUID},
+		OrgID:           utils.Ptr(orgID),
+		UseLatest:       utils.Ptr(true),
+		Arch:            utils.Ptr(config.X8664),
+		Version:         utils.Ptr(config.El8),
+	}
+	tempResp, err := s.dao.Template.Create(ctx, reqTemplate)
+	assert.NoError(s.T(), err)
+	s.updateTemplateContentAndWait(orgID, tempResp.UUID, []string{repoResp.UUID})
+
+	// Update the "red hat" repo with a new URL and snapshot so there are two snapshots
+	opts := serveRepoOptions{
+		port:         "30124",
+		path:         "/" + strings.Split(url, "/")[3] + "/",
+		repoSelector: "frog",
+	}
+	url2, cancelFunc, err := ServeRandomYumRepo(&opts)
+	require.NoError(s.T(), err)
+	defer cancelFunc()
+
+	_, err = s.dao.RepositoryConfig.Update(ctx, config.RedHatOrg, repoResp.UUID, api.RepositoryUpdateRequest{URL: &url2})
+	assert.NoError(s.T(), err)
+
+	fetch, err := s.dao.RepositoryConfig.Fetch(ctx, config.RedHatOrg, repoResp.UUID)
+	assert.NoError(s.T(), err)
+	uuidStr, err = uuid2.Parse(fetch.RepositoryUUID)
+	assert.NoError(s.T(), err)
+
+	snapUUID := s.snapshotAndWait(taskClient, *repoResp, uuidStr, true)
+
+	// Run update-latest-snapshot and verify template is using second snapshot
+	s.updateLatestSnapshotAndWait(config.RedHatOrg, repoResp.UUID)
+
+	tempResp, err = s.dao.Template.Fetch(ctx, orgID, tempResp.UUID, false)
+	assert.NoError(s.T(), err)
+	require.NotNil(s.T(), tempResp.Snapshots)
+	require.NotNil(s.T(), tempResp.Snapshots[0])
+	assert.Equal(s.T(), snapUUID, tempResp.Snapshots[0].UUID)
```

Comment:
Its probably worth adding a fetch of the actual rpm from the 2nd snapshot  (or listing of a letter directory that wouldn't be there in the first snapshot) to verify pulp was updated properly.